Newsletters: add 197 (2022-04-27)

harding opened the pull request:
- [x] Lede, merge summaries, update RCs/releases, add topics @harding 
- [x] StackExchange section @bitschmidty 

Note, the text mentions a thing I posted, so the usual rule applies where I recurse myself from making the ultimate decisions about major edits to that section (or the decision to remove the section if people think it shouldn't be included).

I also have strong opinions about other aspects of the CTV news item, so please please please tell me if you think I'm being biased or unfair in any way.  Thank you for reviewing!

adamjonas commented on _posts/en/newsletters/2022-04-27-newsletter.md:
```diff
@@ -0,0 +1,212 @@
+  - Anthony Towns [analyzed][towns ctv signet] transactions on the CTV
+    [signet][topic signet].  Almost all looked like they may have been
+    constructed with the same software ([Sapio][]), possibly indicating
+    lack of public exploration of CTV.  He further noted that changing
```
```suggestion
    a lack of public exploration of CTV.  He further noted that changing
```

adamjonas commented on _posts/en/newsletters/2022-04-27-newsletter.md:
```diff
@@ -0,0 +1,212 @@
+    justify] the risk".
+
+  - Matt Corallo also [argued][corallo ctv cost] that changing consensus
+    creates significant costs for everyone so we should only attempt
```
```suggestion
    creates significant costs for everyone, so we should only attempt
```

adamjonas commented on _posts/en/newsletters/2022-04-27-newsletter.md:
```diff
@@ -0,0 +1,212 @@
+    later using bare CTV.
+
+    O'Connor's discussion about the address limitation was
+    [mentioned][towns bare] on list by Towns.  O'Connor
```
```suggestion
    [mentioned][towns bare] on the list by Towns.  O'Connor
```

adamjonas commented on _posts/en/newsletters/2022-04-27-newsletter.md:
```diff
@@ -0,0 +1,212 @@
+    Harding suggested temporarily adding CTV to Bitcoin for five years,
+    gathering data during that time about how people used it, and then
+    automatically disabling it unless Bitcoin users five years from now
+    decided it was worth keeping.  No respondents were in favor of this
```
This is my highlight of the newsletter.

bitschmidty commented:
I added two commits for the stack exchange section. The second one adds a (I think) nice image which is creative commons and I have email permission to use from the website author. But we can drop easily if anyone thinks it doesn’t add value

adamjonas commented on _posts/en/newsletters/2022-04-27-newsletter.md:
```diff
@@ -0,0 +1,240 @@
+  summary was being written.  Optech recommends those with strong
+  feelings about the subject---whether in favor or opposed to activating
+  CTV in the next few months---read through the discussions and consider
+  your next actions.  Any person or service provider who doesn't want to
```
Change to `their` to be consistent with those
```suggestion
  their next actions.  Any person or service provider who doesn't want to
```

adamjonas reviewed (commented):
I really enjoyed the CTV write up. No substantial suggestions here, most just readability stuff.

jamesob commented on _posts/en/newsletters/2022-04-27-newsletter.md:
```diff
@@ -0,0 +1,240 @@
+  - Anthony Towns [analyzed][towns ctv signet] transactions on the CTV
+    [signet][topic signet].  Almost all looked like they may have been
+    constructed with the same software ([Sapio][]), possibly indicating
+    lack of public exploration of CTV.  He further noted that changing
```
FWIW, there has since been activity on signet using a modified version of the simple-ctv-vault software: https://twitter.com/fiatjaf/status/1517836181240782850

Xekyo commented on _posts/en/newsletters/2022-04-27-newsletter.md:
```diff
@@ -0,0 +1,240 @@
+
+- [Bitcoin Core 23.0][] is a candidate for the next major
+  version of this predominant full node software.  The [release
+  notes][bcc23 rn] list multiple improvements.  FIXME:harding:improve
```
We should definitely mention that 23.0 creates Taproot descriptors for new wallets. :monocle_face:

Xekyo commented on _posts/en/newsletters/2022-04-27-newsletter.md:
```diff
@@ -0,0 +1,240 @@
+    gathering data during that time about how people used it, and then
+    automatically disabling it unless Bitcoin users five years from now
+    decided it was worth keeping.  No respondents were in favor of this
+    approach, with most of them arguing either that the costs of this
+    approach were too high or the benefits too low.
```
I thought an interesting point in the responses here was that it cannot be fully removed because nodes will need to be able to validate the transactions in that period of the blockchain. Perhaps that is covered in "costs too high" already, but if we were to elaborate, that would be the point I'd include.

Xekyo reviewed (commented):
I liked the CTV coverage, it has a good selection of points highlighted from the responses that focus on relevant issues.

While the mailing list is important and I like our focus on covering only content from it, I think it could be mentioned that the upcoming activation attempt sparked debates and discussions across many venues, not just on the mailing list.

jamesob commented on _posts/en/newsletters/2022-04-27-newsletter.md:
```diff
@@ -0,0 +1,240 @@
+  ctv-st] to the Bitcoin-Dev mailing list his plan to release software
+  that will allow miners to begin signaling whether they intend to
+  enforce the [BIP119][] rules for the proposed
+  [OP_CHECKTEMPLATEVERIFY][topic op_checktemplateverify] (CTV) opcode.
```
Worth noting maybe that Jeremy has indicated (in a public Telegram channel and elsewhere) that he plans to amend the ST parameters that he initially mentioned in the blog post:
- https://twitter.com/JeremyRubin/status/1518477022439247872
- https://t.me/op_ctv/15562

michaelfolkson replied:
I know you're not Jeremy's minder James but as someone who seems to support an upcoming CTV soft fork activation attempt and presumably a participant in the CTV discussions I'm staggered you are comfortable with activation params being dictated on a whim by Jeremy via Twitter and messaging apps. I think even if I was supportive of a soft fork and it had community consensus that would set off alarm bells for me. Regardless I agree with @harding below that the mailing list should be informed. 

Ideally Optech would have covered discussions on the mailing list in previous months about the dangers of attempting soft forks without community consensus but for some reason it has totally ignored them. I'm not quite sure why that was. Presumably for the same reasons why it ignored the UASF client for Taproot and it is ignoring the URSF discussions today. Frankly I think it is irresponsible. But I know it won't make any difference so I won't comment further.

harding replied:
@michaelfolkson 

> Ideally Optech would have covered discussions on the mailing list in previous months about the dangers of attempting soft forks without community consensus but for some reason it has totally ignored them.

We haven't ignored them: https://bitcoinops.org/en/newsletters/2022/03/23/#speedy-trial-discussion

> I won't comment further.

Thanks, I appreciate that.

harding commented:
@Xekyo 

> While the mailing list is important and I'm fine with only covering content from it, I think it could be mentioned that the upcoming activation attempt sparked debates and discussions across many venues, not just on the mailing list.

Sure, I have no problem mentioning that conversations happened elsewhere (and we already mention the chat on bitcoin-wizards).  However, I encourage any of us who learn information useful to the discussion in other venues to relay that info to the mailing list so that it's accessible to everyone watching and participating in the global conversation, as well as being easy to find for future researchers trying to understand why we made the decisions we did.

jamesob commented on _posts/en/newsletters/2022-04-27-newsletter.md:
```diff
@@ -0,0 +1,240 @@
+    LN channel state updates.
+
+  - James O'Beirne [suggested][obeirne benchmark] that his
+    [simple vault][] design could be used as a benchmark for evaluating
```
Cool news, either for here or maybe next issue, that @darosior has adapted this code to work with APO: https://twitter.com/darosior/status/1518961471702642689

jamesob commented:
Thanks for another great summary of a hard-to-summarize issue, @harding!

harding commented:
Made edits for all feedback (thanks so much @adamjonas @jamesob @Xekyo !), added missing content, and added lots of topic links this week.  Reviewed the B.SE section---which is really nice this month!, thanks @bitschmidty !

I'm especially pleased with how this issue turned out, thanks everyone!

bitschmidty commented:
ACK 7f61c5ca814a2cb8da6b6382996c767b4bcc5223

bitschmidty commented:
Great summaries of the CTV activation discussions (and great newsletter overall!), @harding thank you! 🚀

michaelfolkson commented:
Indeed. As always the technical summaries are exceptional. I have strong disagreements on what you are bringing focus to and what you're not but that is the only criticism I can have with this newsletter. It is a wonderful asset for the community. Thanks for getting this out at a stressful time for all of us.
